### PR TITLE
fix: add package1:version:list command, UTs, NUTs

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
adds `package1:version:list` command

### What issues does this PR fix or reference?
requires https://github.com/forcedotcom/packaging/pull/9
@W-10833181@

### CODE REVIEW

**Review by peternhale — APPROVED** [review]

File: src/commands/force/package1/beta/version/list.ts [hunk 1]
```diff
@@ -15,18 +16,37 @@ export class Package1VersionListCommand extends SfdxCommand {
...
+          throw new SfError(messages.getMessage('packageIdInvalid'));
+        }
+      },
     }),
   };
 
-  public async run(): Promise<unknown> {
-    process.exitCode = 1;
-    return Promise.resolve('Not yet implemented');
+  public async run(): Promise<Package1Display[]> {
```
Comment (shetzel) [inline src/commands/force/package1/beta/version/list.ts:35]: What type does this return?

File: test/commands/force/package1/versionList.nut.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import { execCmd, TestSession } from '@salesforce/cli-plugins-testkit';
+import { OrgConfigProperties } from '@salesforce/core';
+import { expect } from 'chai';
+import { Package1Display } from '@salesforce/packaging';
+
+let session: TestSession;
+let usernameOrAlias: string;
+let packageId: string;
+
+// TODO: na40 required as DevHub
```
Comment (shetzel) [inline test/commands/force/package1/versionList.nut.ts:18]: If we don't do something about this todo now, it's unlikely to ever be done.

File: test/commands/force/package1/versionDisplay.nut.ts [hunk 3]
```diff
@@ -42,31 +42,31 @@ describe('package1:version:display', () => {
...
     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
     const output = execCmd(command, { ensureExitCode: 0 }).shellOutput.stdout as string;
     expect(output).to.contain('No results found');
   });
 
-  it('should validate packageversionid flag (too short)', function () {
+  it('should validate packageversionid flag (too short)', () => {
     // fake package ID - too short
     const command = `force:package1:beta:version:display -i 04t46000001Zfa -u ${usernameOrAlias}`;
     // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```
Comment (shetzel) [inline test/commands/force/package1/versionDisplay.nut.ts:56]: Not a huge deal, but we should be able to provide some typing to satisfy eslint right?

**Review by shetzel — APPROVED** [review]
Nothing blocking, but I think addressing them would improve things a bit.

**Pull request merged**